resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/submit/Client.scala

```diff
@@ -81,6 +81,9 @@ private[spark] class Client(
 
   private val driverJavaOptions = submissionSparkConf.get(
     org.apache.spark.internal.config.DRIVER_JAVA_OPTIONS)
+  private val isKerberosEnabled = submissionSparkConf.get(KUBERNETES_KERBEROS_SUPPORT)
+  private val maybeSimpleAuthentication =
+    if (isKerberosEnabled) Some(s"-D$HADOOP_SECURITY_AUTHENTICATION=simple") else None
```

Comment:
simple

Reply:
Why is this the case? The intuition is to set this to `kerberos`.

Reply:
Right. The same question was asked for the executor side as well. Copying my answer below. I'd love to hear your opinions about this:

> Excellent question.
>
> If this is set to "kerberos", then UserGroupInformation code crashes complaining it cannot read kerberos config files like /etc/krb5.conf. So we want to prevent that by suppressing the code path.
>
> As an alternative, we could create a config map containing /etc/krb5.conf and mount it in the driver and executor pods. But that seems to be an overkill.
>
> Now, there is another question. How can setting it to "simple" allows the driver and executor to access secure HDFS? It works because the driver and executor need only the delegation token to access secure HDFS. i.e. They don't need to sign on to Kerberos on their own.
>
> This is counter-intuitive and hard to explain. I am open to suggestions to make this part easier to read. Maybe we can call the associated variable like maybeTokenOnlyAuthentication.

Reply:
Fine as long as we document this.